### DISCUSSION TITLE
Call a query inside getAuth function to validate token

### DISCUSSION THREAD

**Discussion by JonasHiltl** [urql-graphql/urql#2016] [Q&A] [answered]
I want to validate the token received from AsyncStorage (React Native) inside th `getAuth` function with a `useQery` call to my server. But because the Client is not above the `getAuth` function in the component tree I get the warning: 
```
Default Client: No client has been specified using urql's Provider.This means that urql will be falling back to defaults including making requests to `/graphql`.
If that's not what you want, please create a client and add a Provider.
```
```typescript
const getAuth = async ({ authState }: getAuthTypes) => {
  if (!authState) {
    const token = await Storage.get('token')
    // validate the stored token with a query to server
    getUserData()
    if (token) {
      setIsAuthenticated(true)
      return { token }
    }
    setIsAuthenticated(false)
    return null
  }
  setIsAuthenticated(false)
  return null
}
```
The function `getUserData` returns the user data only if the token is valid, the token should be supplied in the auth header, but I'm not sure if there is a way to call a query in the `getAuth` function and also provide some fetch options with auth header set to the token.
Anybody got an idea on how to takle this problem?

**Comment by JoviDeCroock** [comment, marked as answer]
useQuery is a hook so it's scoped to components, you could use your urql client instance however querying will result in hitting the auth-exchange again which makes it a loop. 😅 doesn't the server throw for invalid tokens?
